### PR TITLE
ENH: Include column for ea comparison in asserters

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

The previous exception message did not include the column which made debugging harder than necessary.

### CODE REVIEW

File: pandas/tests/util/test_assert_frame_equal.py [hunk 1]
```diff
@@ -366,3 +366,33 @@ def test_assert_frame_equal_check_like_categorical_midx():
...
+
+    df1 = DataFrame({"a": pd.Series([pd.NA, 1], dtype="Int64")})
+    df2 = DataFrame({"a": pd.Series([pd.NA, 2], dtype="Int64")})
+
+    msg = r'DataFrame.iloc\[:, 0\] \(column name="a"\) values are different'
+    with pytest.raises(AssertionError, match=msg):
+        tm.assert_frame_equal(df1, df2)
+
+    with pytest.raises(AssertionError, match=msg):
+        tm.assert_frame_equal(df1, df2, check_exact=True)
```
Comment (gfyoung) [inline pandas/tests/util/test_assert_frame_equal.py:388]: Two different setups call for two different tests I think

Comment (phofl) [reply pandas/tests/util/test_assert_frame_equal.py:391]: Split

**Comment by gfyoung** [comment]
LGTM, not sure what's going on with GH Actions

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**